### PR TITLE
[lkp-tests PATCHv1] Fix tbench issues

### PR DESCRIPTION
**Pull request by bssrikanth**
This commit:
 1. updates tbench pkg dependecies
 2. update error handling
 3. updates stat parsing
 4. introduces iterations
 5. checks in UT cases for tbench and dbench

### CODE REVIEW

**Comment by bssrikanth** [comment]
This addresses installation issue which was raised in #191 along with few other changes such as introducing iterations, error handling and UT.

**Comment by bssrikanth** [comment]
@rli9 request your review comments.

File: distro/adaptation/centos [hunk 1]
```diff
@@ -131,7 +131,7 @@ libssl: openssl-devel
 libssl-dev: openssl-devel
 libstdc++6: libstdc++.x86_64
 libstdc++-7-dev: libstdc++-devel.x86_64 libstdc++-static.x86_64
-libtirpc: libtirpc1
+libtirpc1: libtirpc
```
Comment (rli9) [inline distro/adaptation/centos:134]: thanks, what is this change for?

File: programs/dbench/parse [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+    stats['clients'] = values[3].to_i
+    stats['procs'] = values[5].to_i
+    stats['max_latency'] << lines[1].split(' ')[0].to_f
+  end
+end
+
+stats.each do |k, v|
+  if k != 'clients' && k != 'procs'
+    puts "#{k}: #{v.average.round(2)}"
+    puts "#{k}_harmonicMean: #{v.harmonic_mean.round(2)}"
```
Comment (rli9) [inline programs/dbench/parse:25]: is it ok to name this as #{k}_harmonic_mean? I found stream also uses harmonicMean and I will change it later to uniform the style.

File: programs/tbench/run [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
 ## it doesn't require a lab of Windows PCs:
+. $LKP_SRC/lib/reproduce-log.sh
 
 [ -n "$direct_server_ips" ] && server=$direct_server_ips
 [ -z "$server" ] && server=127.0.0.1
 [ -n "$runtime" ] || runtime=600
-server=${server%% *}
+[ -n "$iterations" ] || iterations=1
+
+log_cmd command -v tbench || die "tbench not found"
```
Comment (rli9) [inline programs/tbench/run:16]: Here can consider to use has_cmd function

**Comment by rli9** [comment]
> @rli9 request your review comments.

Sorry for the late, just add some comments

File: distro/adaptation/centos (hunk 1 above)
Comment (bssrikanth) [reply distro/adaptation/centos:134]: I did that to overcome below error while insalling tbench on RHEL 8.x/9:
```
Use: /home/LKP/lkp-tests/distro/installer/redhat install libsmbclient libtirpc1 libtirpc-devel rpcgen libsmbclient-devel libxslt popt-devel samba-devel zlib-devel
Updating Subscription Management repositories.
Last metadata expiration check: 0:12:49 ago on Tue 07 Nov 2023 11:09:31 PM CST.
Package libsmbclient-4.16.4-101.el9.x86_64 is already installed.
No match for argument: libtirpc1
Package libxslt-1.1.34-9.el9.x86_64 is already installed.
Package zlib-devel-1.2.11-34.el9.x86_64 is already installed.
Error: Unable to find a match: libtirpc1
Cannot install some packages of tbench depends
```
But now I realize this change was not necessary instead we can just remove that dependency, as I have included libtirpc-devel to take care of libirpc library dependency. I tested it out and install works fine on RHEL. Will send that change in revised push if you are fine with this way.

File: programs/tbench/run (hunk 3 above)
Comment (bssrikanth) [reply programs/tbench/run:16]: Sure will do

File: programs/dbench/parse (hunk 2 above)
Comment (bssrikanth) [reply programs/dbench/parse:25]: Sure

**Comment by bssrikanth** [comment]
@rli9 thanks for your time. I have incorporated review comments, below is the UT result
```
rake spec spec=stats
PWD = /home/LKP/lkp-tests
ENV['LKP_SRC'] = /home/LKP/lkp-tests
/usr/bin/ruby -I/usr/share/gems/gems/rspec-core-3.11.0/lib:/usr/share/gems/gems/rspec-support-3.11.0/lib /usr/share/gems/gems/rspec-core-3.11.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/stats_spec.rb
......................................................................................................................................................................................................................................................................................................................................................................................................

Finished in 1 minute 8.07 seconds (files took 0.23046 seconds to load)
390 examples, 0 failures

```

Please let me know if you have any suggestions.

**Comment by rli9** [comment]
thanks for the patch

**Pull request merged**